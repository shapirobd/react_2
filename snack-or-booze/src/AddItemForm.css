.AddItemForm {
	text-align: left;
}

.AddItemForm input {
	display: block;
	width: 100%;
	margin-bottom: 10px;
}

.AddItemForm label {
	margin-bottom: 3px;
}

.AddItemForm button {
	border: none;
	background-color: white;
	margin: 10px;
	padding: 5px 10px;
	border-radius: 0.5rem;
	color: black;
	text-decoration: none;
}

.AddItemForm button:hover {
	background-color: gray;
	color: white;
}
