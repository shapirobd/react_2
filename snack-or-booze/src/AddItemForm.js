import React from "react";
import "./AddItemForm.css";

// COMPONENT - form to add new drink/snack
const AddItemForm = ({ handleSubmit, handleChange }) => {
	return (
		<form className="AddItemForm" onSubmit={handleSubmit}>
			<label htmlFor="name">Name</label>
			<input type="text" name="name" onChange={handleChange} />

			<label htmlFor="description">Description</label>
			<input type="text" name="description" onChange={handleChange} />

			<label htmlFor="recipe">Recipe</label>
			<input type="text" name="recipe" onChange={handleChange} />

			<label htmlFor="serve">Serve</label>
			<input type="text" name="serve" onChange={handleChange} />

			<button>Submit</button>
		</form>
	);
};

export default AddItemForm;
