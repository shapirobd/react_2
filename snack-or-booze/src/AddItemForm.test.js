import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import AddItemForm from "./AddItemForm";

test("it should render without crashing", () => {
	render(
		<MemoryRouter>
			<AddItemForm />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<AddItemForm />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
