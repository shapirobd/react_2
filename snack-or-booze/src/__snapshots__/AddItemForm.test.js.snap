// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <form
    class="AddItemForm"
  >
    <label
      for="name"
    >
      Name
    </label>
    <input
      name="name"
      type="text"
    />
    <label
      for="description"
    >
      Description
    </label>
    <input
      name="description"
      type="text"
    />
    <label
      for="recipe"
    >
      Recipe
    </label>
    <input
      name="recipe"
      type="text"
    />
    <label
      for="serve"
    >
      Serve
    </label>
    <input
      name="serve"
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
